<?xml version="1.0"?>
<launch>
  <arg name="name" default="bebop"/>
  <arg name="ip" default="192.168.42.1" />            <!-- IPv4 address of your drone -->
  <arg name="drone_type" default="bebop2" />          <!-- available drone types: bebop1, bebop2 -->
  <arg name="config_file" default="$(find teamsannio_med_control)/config/defaults.yaml" />
  <arg name="camera_info_url" default="package://teamsannio_med_control/data/$(arg drone_type)_camera_calib.yaml" />
  
     <!-- Launch the controller  -->
     <node name="position_controller_with_bebop_node" pkg="teamsannio_med_control" type="position_controller_with_bebop_node" output="screen"> 
       <rosparam command="load" file="$(find teamsannio_med_control)/resource/controller_bebop.yaml" />
       <rosparam command="load" file="$(find teamsannio_med_control)/resource/bebop.yaml" />
       <rosparam command="load" file="$(find teamsannio_med_control)/resource/EKF_matrix.yaml" />
       <remap from="/pose" to="/bebop/optitrack/pose" />
       <remap from="/velocity" to="/bebop/optitrack/velocity" />
       <remap from="/command/trajectory" to="/bebop/command/trajectory" />
       <remap from="/referenceAngles" to="/bebop/referenceAngles" />
       <remap from="/filteredOutput" to="/bebop/filteredOutput" />
       <remap from="/smoothedTrajectory" to="/bebop/smoothedTrajectory" />
       <remap from="/cmd_vel" to="/bebop/cmd_vel" />
       <remap from="/bebopAngles" to="/bebop/bebopAngles" />
      </node>

   <group ns="$(arg name)">

   <!-- Launch the trajectory generator -->
    <node name="hovering_example_with_bebop" pkg="teamsannio_med_control" type="hovering_example_with_bebop" output="screen"/>

    <!-- Launch the trajectory plot -->
    <node name="position_plot" pkg="rqt_plot" type="rqt_plot" args="--clear-config /$(arg name)/odometry/pose/pose/position" output="screen" />
    <node name="quaternion_to_rpy" pkg="teamsannio_med_control" type="quaternion_to_rpy" output="screen" >
       <remap from="/odometry" to="/bebop/odometry" />
       <remap from="/orientation_rpy" to="/bebop/orientation_rpy" />
    </node>
    <node name="orientation_plot" pkg="rqt_plot" type="rqt_plot" args="--clear-config /$(arg name)/orientation_rpy" output="screen" />

    <!-- Launch the reference angles plot -->
    <node name="reference_angles_plot" pkg="rqt_plot" type="rqt_plot" args="--clear-config /$(arg name)/referenceAngles/pose/pose/position" output="screen"  />
    <node name="uXY_plot" pkg="rqt_plot" type="rqt_plot" args="--clear-config /$(arg name)/referenceAngles/twist/twist/linear" output="screen" />

    <!-- Launch the smoothed trajectory plot -->
    <node name="smoothed_plot" pkg="rqt_plot" type="rqt_plot" args="--clear-config /$(arg name)/smoothedTrajectory/pose/pose/position" output="screen" />

    <!-- Bebop angles -->
    <node name="bebop_angles_plot" pkg="rqt_plot" type="rqt_plot" args="--clear-config /$(arg name)/bebopAngles/pose/pose/position" output="screen"  />

   </group>

</launch>

